packages/commonwealth/server/util/cosmosCache.ts

```diff
@@ -0,0 +1,113 @@
+import type { Request, RequestHandler } from 'express';
+import { cacheDecorator } from 'common-common/src/cacheDecorator';
+
+export const cosmosCacheRPC = (duration: number): RequestHandler => {
+  const cacheFunc = (ttl: number) =>
+    cacheDecorator.cacheMiddleware(ttl, cosmosRPCKeyGenerator);
+
+  // RPC specific codes from cosmJS requests
+  const activeCodes = ['0801', '0802'];
+  const completedCodes = [
+    '0803',
+    '0804',
+    '0805',
+    '0803220a0a080000000000000087',
+    '0803220a0a080000000000000100',
+    '0803220a0a080000000000000189',
+  ];
+
+  return async function cacheIfCacheworthyRPC(req, res, next) {
+    let body = req.body;
+    try {
+      body = JSON.parse(req.body);
+    } catch (err) {
+      console.log('Error parsing body: ', err);
+    }
+
+    const params = JSON.stringify(body?.params);
+    const method = body?.method;
+
```

Comment:
@jnaviask While I work on tests, it would be good to get your feedback on the rules in this file. 

Mostly I'm deciding whether to cache or not, and how long the TTL should be.

Reply:
Few notes:
1. We can cache completed proposals for longer, as they will ONLY EVER change when an active proposal becomes completed.
2. Would like to understand concerns re caching block, if any.
3. What's the reasoning behind 15 minute pool cache? Seems okay to me, but was wondering why we landed there.

Reply:
@jnaviask 

1. I'm happy with caching completed proposals longer. I was just thinking about the case where a proposal transitions into completed, but doesn't show up in the UI until the cache expires. If we don't mind that, we can cache longer.
2. Caching block makes sense to me for a standard block time (6 seconds). After that, we refresh the cached block.
    - There may be a concern where a user gets a stale (by 1 block) block response. I haven't fully worked through whether there is any downside to this. Upside is performance.
3. No reasoning on 15minutes, except wanting something that will stick long enough to give us performance benefits. Happy to adjust.

Reply:
@jnaviask Also, thinking we should cache status checks for 1 block as well. That way, all of our standard requests get potential benefit from caching, if only for 6 seconds.

Reply:
This all sounds good to me. Re completed proposals, my worry is that they are the most heavyweight thing we cache, and are loaded somewhat infrequently by users -- so we may not see a benefit from caching them unless we use a longer TTL.

Your concern is also valid. We may want to augment the logic in the future if possible. The ideal case, we cache completed proposals for a long time, and invalidate them on certain conditions (i.e. the state of active proposals changes). I don't know if that's possible now, so we should find a good compromise between longevity wins and UI accuracy.

Reply:
@jnaviask How about 30 minutes for completed proposals? More?

On `status`, I will put this on a short cache.

Reply:
Actually, how about 60 minutes?

Reply:
One question @mhagel is there something we can cache with no-expiry?

Duration=0 will cache without TTL
Thats an option too if not used already

Reply:
There is one case where we are skipping the cache: `tx` responses. These are when a user makes a transaction (ie vote) so they should not be cached. We could use the 0 duration, but I chose to just bypass the cache middleware in this case.

Reply:
As discussed, completed proposals set to 15 min

Reply:
duration 0 doesn't skip caching (edited) 
instead its for caching forever

if we have any caching forever case please pass 0, thanks!